src/classes/DateLiteralConstant.cls

```diff
@@ -0,0 +1,225 @@
+public class DateLiteralConstant {
+    private String LAST_N = 'LAST_N_{0}: {1}';
```

Comment:
I'm actually not sure.... can you make these 2 strings 'final' as well, and still use String.format() later? If you can, then I'd say make them 'final' as well.